base/services/umpnpmgr/install.c

```diff
@@ -366,7 +367,10 @@ DeviceInstallThread(LPVOID lpParameter)
 
     while (TRUE)
     {
-        ListEntry = InterlockedPopEntrySList(&DeviceInstallListHead);
+        WaitForSingleObject(hDeviceInstallListMutex, INFINITE);
+        ListEntry = (IsListEmpty(&DeviceInstallListHead)
+                        ? NULL : /*RemoveHeadList*/RemoveTailList(&DeviceInstallListHead));
```

Comment:
@JoachimHenze : Please retest with this modification:
```suggestion
                        ? NULL : RemoveHeadList(&DeviceInstallListHead));
```

Reply:
Ship it :)